packages/jupyter-ai/src/components/chat-settings.tsx

```diff
@@ -29,6 +32,50 @@ enum ChatSettingsState {
   Success
 }
 
+function getProviderId(globalModelId: string | null) {
+  if (!globalModelId) {
+    return null;
+  }
+
+  return globalModelId.split(':')[0];
```

Comment:
Should we validate that the globalModelId has exactly 2 components? Otherwise, if an abbreviated global model ID is used, that might be misinterpreted as a provider ID.

Reply:
The list of models returned from the ListProviders API should always return valid global model IDs. If we need further validation, this should be done in the backend instead.